src/topology/filter.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2022 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import order.filter.lift
+import topology.separation
+import data.set.intervals.monotone
+
+/-!
+# Topology on the set of filters on a type
+
+In this file introduce topology on `filter α`. It is generated by the sets
+`set.Iic (𝓟 s) = {l : filter α | s ∈ l}`, `s : set α`. A set `s : set (filter α)` is open if and
+only if it is a union of a family of these basic open sets, see `filter.is_open_iff`.
+
+This topology has the following important properties.
+
+* If `X` is a topological space, then the map `𝓝 : X → filter X` is an inducing map.
+
+* In particular, it is a continuous map, so `𝓝 ∘ f` tends to `𝓝 (𝓝 a)` whenever `f` tends to `𝓝 a`.
+
+* If `X` is an ordered topological space with order topology and no max element, then `𝓝 ∘ f` tends
+  to `𝓝 filter.at_top` whenever `f` tends to `filter.at_top`.
+
+* It turns `filter X` into a T₀ space and the order on `filter X` is the dual of the
+  `specialization_order (filter X)`.
```

Comment:
I never used the specialization order outside of mathlib, so I don't feel qualified to change its definition.

Reply:
I would expect people to mostly use the `⤳` anyway, which goes in the right way, so I don't think changing it would cause much trouble. But we should ask on Zulip before making such a change, and it can definitely wait for another PR.